scio-tensorflow/src/test/scala/com/spotify/scio/io/TFTapTest.scala

```diff
@@ -40,4 +56,42 @@ class TFTapTest extends TapSpec {
     }
   }
 
+  it should "support saveAsTfExampleFile" in {
+    import com.spotify.scio.tensorflow._
+    val examples = getDummyExample
+    import org.apache.beam.sdk.io.TFRecordIO.{CompressionType => CType}
+    for (compressionType <- Seq(CType.NONE, CType.ZLIB, CType.GZIP)) {
+      val dir = tmpDir
+      val sc = ScioContext()
+      val featureSpec = sc.parallelize(Seq(Seq("f1", "f2")))
```

Comment:
This `Seq(Seq())` and several other places may be confusing. Anyway to simplify it?
We can't overload `saveAsTfExampleFile` due to the default args. What about making the argument `featureSpec: M` where `[M: FeatureSpecType]`? You can then have 3 instances where `M` is a case class, `Seq[String]` or `SCollection[Seq[String]]`.

By doing this, you can probably also remove the following and pass `WordCountFeatures.type` as `featureSpec` or something like that.
`val featureNameSpec: Seq[String] = FeatureNameFeatureSpec[WordCountFeatures]`

Thoughts? Too complicated?